test/e2e/network/loadbalancer.go

```diff
@@ -298,8 +319,19 @@ var _ = common.SIGDescribe("LoadBalancers", func() {
 		ns2 := f.Namespace.Name // LB1 in ns2 on TCP
 		framework.Logf("namespace for TCP test: %s", ns2)
 
-		nodeIP, err := e2enode.PickIP(cs) // for later
+		node, err := e2enode.GetRandomReadySchedulableNode(cs)
 		framework.ExpectNoError(err)
+		var nodeIP string
+		ips := e2enode.GetAddresses(node, v1.NodeExternalIP)
+		if len(ips) > 0 {
+			nodeIP = ips[0]
+		} else {
+			ips = e2enode.GetAddresses(node, v1.NodeInternalIP)
+			if len(ips) == 0 {
+				framework.Failf("Node %v without IP address", node)
+			}
+			nodeIP = ips[0]
+		}
```

Comment:
I feel like there must be a helper function for this.

But also, this is further committing us to "pods must be able to reach ExternalIPs", re #113799

Reply:
this is kind of different, since this test requires to poll a node from the place where the e2e.test binary is running ... all the other tests probe from within the cluster, we can not do this since we want to exercise the loadbalancer path

Let me refactor into a helper

Reply:
> Let me refactor into a helper

No, I meant "I feel like there must _already_ be a helper function for this".

But it seems like there is not; you have three constraints ("use a randomly-selected node, not the first node", "use an IP of a specific family", and "use either an ExternalIP or an InternalIP, depending on what exists"). And there are helpers for most combinations of 2 of those constraints, but none that incorporates all 3.